tests/system/providers/google/cloud/compute/example_compute_igm.py

```diff
@@ -0,0 +1,249 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG that:
+* creates a copy of existing Instance Template
+* updates existing template in Instance Group Manager
+
+"""
+
+import os
+from datetime import datetime
+
+from airflow import models
+from airflow.models.baseoperator import chain
+from airflow.operators.bash import BashOperator
+from airflow.providers.google.cloud.operators.compute import (
+    ComputeEngineCopyInstanceTemplateOperator,
+    ComputeEngineDeleteInstanceGroupManagerOperator,
+    ComputeEngineDeleteInstanceTemplateOperator,
+    ComputeEngineInsertInstanceGroupManagerOperator,
+    ComputeEngineInsertInstanceTemplateOperator,
+    ComputeEngineInstanceGroupUpdateManagerTemplateOperator,
+)
+from airflow.utils.trigger_rule import TriggerRule
+
+ENV_ID = os.environ.get('SYSTEM_TESTS_ENV_ID')
+PROJECT_ID = os.environ.get('SYSTEM_TESTS_GCP_PROJECT')
+
+LOCATION = 'europe-west1-b'
+REGION = 'europe-west1'
+SHORT_MACHINE_TYPE_NAME = 'n1-standard-1'
+DAG_ID = 'cloud_compute_igm'
+
+# [START howto_operator_compute_template_copy_args]
+TEMPLATE_NAME = 'instance-template-igm-test'
+NEW_TEMPLATE_NAME = 'instance-template-test-new'
+
+INSTANCE_TEMPLATE_BODY = {
+    "name": TEMPLATE_NAME,
+    "properties": {
+        "machine_type": SHORT_MACHINE_TYPE_NAME,
+        "disks": [
+            {
+                "auto_delete": True,
+                "boot": True,
+                "device_name": TEMPLATE_NAME,
+                "initialize_params": {
+                    "disk_size_gb": "10",
+                    "disk_type": "pd-balanced",
+                    "source_image": "projects/debian-cloud/global/images/debian-11-bullseye-v20220621",
+                },
+            }
+        ],
+        "network_interfaces": [{"network": "global/networks/default"}],
+    },
+}
+
+NEW_DESCRIPTION = 'Test new description'
+INSTANCE_TEMPLATE_BODY_UPDATE = {
+    "name": NEW_TEMPLATE_NAME,
+    "description": NEW_DESCRIPTION,
+    "properties": {"machine_type": "n1-standard-2"},
+}
+# [END howto_operator_compute_template_copy_args]
+
+# [START howto_operator_compute_igm_update_template_args]
+INSTANCE_GROUP_MANAGER_NAME = 'instance-group-test'
+INSTANCE_GROUP_MANAGER_BODY = {
+    "name": INSTANCE_GROUP_MANAGER_NAME,
+    "base_instance_name": INSTANCE_GROUP_MANAGER_NAME,
+    "instance_template": f'global/instanceTemplates/{NEW_TEMPLATE_NAME}',
+    "target_size": 1,
+}
+
+SOURCE_TEMPLATE_URL = os.environ.get(
```

Comment:
Do we need to retrieve SOURCE_TEMPLATE_URL if we can use default to ``https://www.googleapis.com/compute/beta/projects/{PROJECT_ID}/global/instanceTemplates/{TEMPLATE_NAME}"``? (we're trying to limit amount of the required env params for running the tests)

Reply:
Done